calculator_test.go

```diff
@@ -2,23 +2,230 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
+func TestCloseEnough(t *testing.T) {
+	testCases := []struct {
+		a, b, roundedResult, tolerance float64
+		want                           bool
+	}{
+		{a: 5, b: 3, roundedResult: 1.6666, tolerance: 0.0001},
+		{a: 2, b: 3, roundedResult: 0.6666, tolerance: 0.001},
+		{a: 1, b: 3, roundedResult: 0.3333, tolerance: 0.01},
+	}
+
+	for _, tc := range testCases {
+
+		result, err := calculator.Divide(tc.a, tc.b)
+		if err != nil {
+			t.Fatalf("%v", err)
+		}
+
+		if !calculator.CloseEnough(tc.roundedResult, result, tc.tolerance) {
```

Comment:
In fact, we don't really need to export this function, or put it in the calculator package, do we? It's purely an internal test helper, if you like.

Reply:
Fixed in df06d53